isis/src/qisis/objs/JigsawSetupDialog/JigsawSetupDialog.cpp

```diff
@@ -1832,15 +1831,90 @@ namespace Isis {
     // up the new bundle observation solve settings
     updateBundleObservationSolveSettings(solveSettings);
 
+    // Loop through the existing solve settings list and determine colors
+    QSet<QString> existingColorNames;
+    qDebug() << "# solve settings: " << solveSettingsList.size();
+    for (auto &solveSettings : solveSettingsList) {
+      qDebug() << "    color: " << solveSettings.color();
+      qDebug() << "    colorName: " << solveSettings.color().name();
+      existingColorNames.insert(solveSettings.color().name());
+    }
+
+    QSet<QString> badColorSet {
+      "#FFB300", // Vivid Yellow
+      "#803E75", // Strong Purple
+      "#FF6800", // Vivid Orange
+      "#A6BDD7", // Very Light Blue
+      "#C10020", // Vivid Red
+      "#CEA262", // Grayish Yellow
+      "#817066", // Medium Gray
+    };
+    QSet<QString> colorSet;
+    for (auto &color : badColorSet) {
+      colorSet.insert(color.toLower());
+    }
+
+    // Change color for new solve settings
+    QColor newColor(Qt::white);
+
+    QColor red(Qt::red);
+    qDebug() << "red hex: " << red.name();
+    red.setNamedColor("#CCCCCC");
+    qDebug() << "red hex2: " << red.name();
+
+    QSet<QString> availableColors = colorSet.subtract(existingColorNames);
```

Comment:
Right now we are limited to seven BOSS colors, but we could expand this to rotate over the colors. Maybe you could maintain an index for the set and modulo by the length to loop over it as many times as you want.

We could combine the current "base color" with an offset to randomly create unique colors.
Here is a link to some good ideas on color generation: http://devmag.org.za/2012/07/29/how-to-choose-colours-procedurally-algorithms/. The offset color generation is under the Algorithms section, in the random offset subsection.

With this idea we would loop through the color set, creating unique colors in a set order. For example, if I had three hues in my "badColorSet": 

Red -> Blue -> Yellow -> Red w/ offset 1 -> Blue w/ offset 1 -> Yellow w/ offset 1 -> Red w/ offset 2...

Reply:
That's a cool idea! But I think that might be a stretch goal for now.